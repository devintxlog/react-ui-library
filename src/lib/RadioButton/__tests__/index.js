import React from 'react'
import { shallow } from 'enzyme'
import RadioButton from '..'

describe('RadioButton', () => {
  let props = {
    labelText: `Example`,
    value: true
  }
  
  //create a shallow wrapper that can be reused
  let wrapper
  //initialize the components
  beforeEach(() => {
    wrapper = shallow(<RadioButton {...props}/>)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('when disabled prop is true', () => {
    it('renders correctly', () => {
      wrapper.setProps({
        disabled: true
      })
      expect(wrapper).toMatchSnapshot()
    })
  })
})