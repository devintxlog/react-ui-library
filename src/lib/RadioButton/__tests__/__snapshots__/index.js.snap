// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton renders correctly 1`] = `
<div
  className="container"
>
  <input
    defaultChecked={false}
    type="radio"
    value={true}
  />
  <div
    className="element"
  />
  <label>
    Example
  </label>
</div>
`;

exports[`RadioButton when disabled prop is true renders correctly 1`] = `
<div
  className="container disabled"
>
  <input
    defaultChecked={false}
    disabled={true}
    type="radio"
    value={true}
  />
  <div
    className="element"
  />
  <label>
    Example
  </label>
</div>
`;
